### PR TITLE
Add runpod sls handler

### PR DESCRIPTION
**Pull request by KAJdev**
Adds the `.runpod` directory with a serverless handler & runpod hub configuration with tests.

# Description

This adds the necessary files to run Axolotl on RunPod serverless, and creates a `hub.json` for RunPod hub metadata, and `tests.json` for automated tests.

In order to deploy to the runpod hub, a release must be created.

Example payload:
```json
{
  "input": {
    "user_id": "user",
    "model_id": "model-name",
    "run_id": "run-id",
    "credentials": {
      "wandb_api_key": "", # add your Weights & biases key. TODO:  you will be able to set this in Enviornment variables.
      "hf_token": "", # add your HF_token. TODO:  you will be able to set this in Enviornment variables.
    },
    "args": {
      "base_model": "NousResearch/Llama-3.2-1B",
      // ... other options
    }
  }
}
```

A full configuration reference can be found in the `.runpod/README.md`

### CODE REVIEW

File: .runpod/src/utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#     """
+#     Currently limited by all possible axolotl args, no defaults
+#     :param input_args: dict of input args
+#     """
+#     all_args = yaml.safe_load(open("config/config.yaml", "r"))
+#     if not input_args:
+#         print("No args provided, using defaults")
+#     else:
+#         all_args.update(input_args)
+#     return all_args
```
Comment (djsaunde) [inline .runpod/src/utils.py]: Can we remove?

File: .runpod/src/train.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+async def train(config_path: str, gpu_id: str = "0", preprocess: bool = True):
+    """
+    Run preprocessing (if enabled) and training with the given config file
+    :param config_path: Path to the YAML config file
+    :param gpu_id: GPU ID to use (default: "0")
+    :param preprocess: Whether to run preprocessing (default: True)
+    
+    """
```
Comment (djsaunde) [inline .runpod/src/train.py]: optional nit: we autogen docs with google-style docstrings; we may want to update to take advantage of that.

File: .runpod/src/utils.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def make_valid_config(input_args):
```
Comment (djsaunde) [inline .runpod/src/utils.py:15]: This seems not to be used anywhere?

File: .runpod/src/utils.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    updated_config_path = "config/updated_config.yaml"
+
+    # Save updated config to new file
+    with open(updated_config_path, "w") as f:
+        yaml.dump(all_args, f)
+
+    return updated_config_path
+
+
+def set_config_env_vars(args: dict):
```
Comment (djsaunde) [inline .runpod/src/utils.py:40]: Ditto for this function.

File: .runpod/README.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+    },
+    "args": {
+      "base_model": "NousResearch/Llama-3.2-1B",
+      // ... other options
+    }
+  }
+}
+```
+
+## Configuration Options
```
Comment (djsaunde) [inline .runpod/README.md:31]: We document configuration options [here](https://docs.axolotl.ai/docs/config.html). I'm not sure we want to duplicate that here and create the need to maintain two different sources of truth. Can we instead link to those docs?

File: .runpod/README.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+# Configuration Options
+
+This document outlines all available configuration options for training models. The configuration can be provided as a JSON request.
+
+## Usage
+
+You can use these configuration Options:
+
+1. As a JSON request body:
```
Comment (djsaunde) [inline .runpod/README.md:11]: I noticed this is the only item in a numbered list. Do you have other usage options or can we remove the numbering?

Also, is there an easy way (e.g., a script) to create this request body from a config YAML file?

File: .runpod/src/config/config.yaml [hunk 7]
```diff
@@ -0,0 +1,577 @@
+# # This is the huggingface model that contains *.pt, *.safetensors, or *.bin files
```
Comment (djsaunde) [inline .runpod/src/config/config.yaml:1]: Can we remove all these commented lines?

**Review by djsaunde — COMMENTED** [review]
This is cool, thanks for putting this together. I left some comments / questions, but I'll defer to @winglian for final approval

**Comment by NanoCode012** [comment]
Hello @KAJdev , thanks for the PR! Could you let me know if there's some way to test this before merge (like pointing to your branch from the main site)?

Secondly, how/where is the automated test ran?

File: .runpod/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,18 @@
+FROM runpod/pytorch:3.10-2.0.0-117
```
Comment (winglian) [inline .runpod/Dockerfile]: It might be better to use the official axolotl images?

File: .runpod/README.md [hunk 9]
```diff
@@ -0,0 +1,335 @@
+<h1>LLM Training- Full finetune, LoRA, QLoRa etc. Llama/Mistral/Gemma</h1>
```
Comment (NanoCode012) [inline .runpod/README.md]:
```suggestion
<h1>LLM Post Training- Full fine-tune, LoRA, QLoRa etc. Llama/Mistral/Gemma and more</h1>
```

File: .runpod/hub.json [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+      "12.7",
+      "12.6",
+      "12.5",
+      "12.4",
+      "12.3",
+      "12.2",
+      "12.1",
+      "12.0",
+      "11.8"
+    ],
```
Comment (NanoCode012) [inline .runpod/hub.json]: We currently build for cuda 12.4, 12.6 (and soon 12.8). Do we need to trim this and below lists to these two versions?

Comment (winglian) [reply .runpod/hub.json]: I pruned this

File: .runpod/README.md
Comment (winglian) [inline .runpod/README.md:1]: Its this documentation specific to users reading this within Axolotl? Or does this README get rendered in the Hub UI?
